src/components/Menu/Menu.test.tsx

```diff
@@ -55,4 +56,25 @@ describe('<Menu />', () => {
 
     expect(screen.queryByTestId('menu-content')).not.toBeInTheDocument();
   });
+
+  it('should allow render prop usage', async () => {
```

Comment:
oh neat, pleasantly surprised that this already works without the extra logic. good idea to add a test 👍

Reply:
this tests HeadlessUI behavior, but does ensure we didn't prevent it from working. Good Test